contractcourt/htlc_success_resolver.go

```diff
@@ -174,6 +176,14 @@ func (h *htlcSuccessResolver) Resolve() (ContractResolver, error) {
 			return nil, fmt.Errorf("quitting")
 		}
 
+		// With the HTLC claimed, we can attempt to settle its
+		// corresponding invoice if we were the original destination.
```

Comment:
We are the original destination, aren't we?

Reply:
Either we're the final destination, or we're just an intermediate node that happend to already go on-chain for the incoming HTLC (the outgoing was pulled off-chain).

Reply:
Ah yes, right.